src/reactor.cc

```diff
@@ -0,0 +1,333 @@
+#include "reactor.h"
+
+#include "boost/shared_ptr.hpp"
+
+namespace tricycle {
+
+Reactor::Reactor(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  fuel_tracker.Init({&tritium_storage}, fuel_limit);
+  blanket_tracker.Init({&blanket}, blanket_limit);
+}
+
+void Reactor::Tick() {
+  if (sufficient_tritium_for_operation) {
+    SequesterTritium();
+    OperateReactor(TBR);
+    blanket_fill_policy.Start();
+    Reactor::RecordStatus("Online", fusion_power);
+  } else {
+    Reactor::RecordStatus("Shut-down", 0);
+  }
+
+  DecayInventory(tritium_storage);
+  DecayInventory(tritium_excess);
+  sequestered_tritium->Decay(context()->time());
+
+  ExtractHelium(tritium_storage);
+  ExtractHelium(tritium_excess);
+
+  if (!tritium_storage.empty() && sufficient_tritium_for_operation) {
+    double surplus = std::max(
+      tritium_storage.quantity() - reserve_inventory, 0.0);
+
+    if (surplus > 0.0) {
+
+      tritium_excess.Push(tritium_storage.Pop(surplus));
+      CombineInventory(tritium_excess);
+
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from storage to excess");
+    }
+  }
+
+  if (context()->time() % blanket_turnover_frequency == 0 && !blanket.empty()) {
+    if (blanket.quantity() >= blanket_turnover) {
+      blanket_excess.Push(blanket.Pop(blanket_turnover));
+      CombineInventory(blanket_excess);
+      RecordOperationalInfo(
+          "Blanket Cycled",
+          std::to_string(blanket_turnover) + "kg of blanket removed");
+    } else {
+      RecordOperationalInfo(
+          "Blanket Not Cycled",
+          "Total blanket material (" + std::to_string(blanket.quantity()) +
+              ") insufficient to extract " +
+              std::to_string(blanket_turnover) + "kg!");
+    }
+  }
+}
+
+void Reactor::Tock() {
+  if (!sufficient_tritium_for_operation) {
+    try {
+      Startup();
+      fuel_startup_policy.Stop();
+      fuel_refill_policy.Start();
+    } catch (const std::exception& e) {
+      RecordOperationalInfo("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+    }
+  }
+
+  CombineInventory(tritium_storage);
+  CombineInventory(blanket);
+
+  RecordInventories(tritium_storage.quantity(), tritium_excess.quantity(), 
+                    sequestered_tritium->quantity(), blanket.quantity(),
+                    blanket_excess.quantity(), helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  fuel_startup_policy
```

Comment:
We should discuss all these policies at a fuel cycle meeting with @nuclearkatie present.  My hunch is that there is an easier way to accomplish what you want.